// @wix/tpa-settings/dist/es/src/contexts/TpaSettingsProvider/TPASettingsProviderForWidget.js
import React from 'react';
import {
    SettingsProvider,
    SettingsConsumer
} from '../SettingsContext';
import {
    StylesProvider,
    StylesConsumer,
} from '../StylesContext';
export var TpaSettingsProviderForWidget = function(_a) {
    var t = _a.t,
        experiments = _a.experiments,
        publicData = _a.publicData,
        isMobile = _a.isMobile,
        isRTL = _a.isRTL,
        isEditorX = _a.isEditorX,
        dimensions = _a.dimensions,
        language = _a.language,
        children = _a.children,
        hostStyle = _a.hostStyle,
        stylesParams = _a.stylesParams,
        styleParamsPerBreakpointMode = _a.styleParamsPerBreakpointMode,
        currentStylesGetter = _a.currentStylesGetter,
        allStylesGetter = _a.allStylesGetter;
    return (React.createElement(SettingsProvider, {
            t: t,
            experiments: experiments,
            publicData: publicData,
            isMobile: isMobile,
            isRTL: isRTL,
            isEditorX: isEditorX,
            language: language
        },
        React.createElement(StylesProvider, {
                userStylesParams: stylesParams,
                currentStylesGetter: currentStylesGetter,
                allStylesGetter: allStylesGetter,
                styleParamsPerBreakpointMode: styleParamsPerBreakpointMode,
                styleParams: hostStyle === null || hostStyle === void 0 ? void 0 : hostStyle.styleParams,
                textPresets: hostStyle === null || hostStyle === void 0 ? void 0 : hostStyle.siteTextPresets,
                siteColors: hostStyle === null || hostStyle === void 0 ? void 0 : hostStyle.siteColors,
                isMobile: isMobile,
                isRTL: isRTL,
                experiments: experiments,
                dimensions: dimensions
            },
            React.createElement(SettingsConsumer, null, function(settings) {
                return (React.createElement(StylesConsumer, null, function(styles) {
                    return settings.ready && styles.ready && children;
                }));
            }))));
};
