// @wix/blocks-widget-services/dist/esm/utils/blocksConfigParser.js
import {
    NotBlocksAppError
} from '../errors/notBlocksAppError';
import {
    extractSearchParams
} from './stringUtils';
export const parseBaseUrlsData = (baseUrls) => {
    const urlParamsStr = extractSearchParams(baseUrls.siteAssets);
    const urlParams = new URLSearchParams(urlParamsStr);
    return {
        siteId: urlParams.get('siteId'),
        metaSiteId: urlParams.get('metaSiteId'),
        revision: urlParams.get('siteRevision'),
    };
};
export const isBlocksAppParams = (appParams) => Boolean(appParams.baseUrls.siteAssets) && 'blocksData' in appParams;
export const isBlocksPreviewAppParams = (appParams) => Boolean(appParams.appData && 'blocksPreviewData' in appParams.appData);
const isBlocksComponent = (comp) => comp.compType === 'STUDIO';
const isValidAppData = (app) => Boolean(app === null || app === void 0 ? void 0 : app.components);
const toAppParams = (app) => {
    if (!isValidAppData(app)) {
        return;
    }
    const blocksComp = app.components.find(isBlocksComponent);
    if (!blocksComp) {
        return;
    }
    return {
        appDefinitionId: app.appId,
        appName: app.name,
        baseUrls: blocksComp.compData.studioComponentData.baseUrls,
        blocksData: {
            siteHeaderUrl: blocksComp.compData.studioComponentData.siteHeaderUrl,
            wixCodeGridId: blocksComp.compData.studioComponentData.wixCodeGridId,
            wixCodeInstanceId: blocksComp.compData.studioComponentData.wixCodeInstanceId,
        },
        appInstanceId: '',
        appRouters: [],
        instance: '',
        instanceId: '',
        isRollout: false,
        url: '',
    };
};
export function parseBlocksConfigFromAppData(app) {
    const appParams = toAppParams(app);
    if (!appParams) {
        return;
    }
    return parseBlocksConfig(appParams);
}
export default function parseBlocksConfig(appParams) {
    var _a, _b, _c, _d;
    if (!isBlocksAppParams(appParams)) {
        throw new NotBlocksAppError();
    }
    const {
        wixCodeInstanceId,
        wixCodeGridId
    } = appParams.blocksData;
    const {
        siteId,
        metaSiteId,
        revision
    } = parseBaseUrlsData(appParams.baseUrls);
    return {
        appDefinitionId: appParams.appDefinitionId,
        code: {
            instanceId: wixCodeInstanceId,
            gridId: wixCodeGridId,
            experimentsQueryString: ((_b = (_a = appParams.appData) === null || _a === void 0 ? void 0 : _a.blocksConsumerData) === null || _b === void 0 ? void 0 : _b.codeExperimentsQueryString) || '', // Should be replaced with a type guard once WBL-2343 is done
        },
        siteId,
        metaSiteId,
        revision,
        codePackagesData: (_d = (_c = appParams.appData) === null || _c === void 0 ? void 0 : _c.blocksConsumerData) === null || _d === void 0 ? void 0 : _d.codePackagesData,
    };
}
export const isPanelControllerConfig = (controllerConfig) => {
    var _a, _b, _c, _d;
    if (!((_b = (_a = controllerConfig === null || controllerConfig === void 0 ? void 0 : controllerConfig.wixCodeApi) === null || _a === void 0 ? void 0 : _a.location) === null || _b === void 0 ? void 0 : _b.url)) {
        return false;
    }
    const url = new URL((_d = (_c = controllerConfig === null || controllerConfig === void 0 ? void 0 : controllerConfig.wixCodeApi) === null || _c === void 0 ? void 0 : _c.location) === null || _d === void 0 ? void 0 : _d.url);
    const sdkVersion = url.searchParams.get('sdkVersion');
    const componentRef = url.searchParams.get('componentRef');
    return sdkVersion !== null && componentRef !== null;
};
