// @wix/yoshi-flow-editor-runtime/build/esm/environment/Widget.js
import {
    isEditorX,
    isADI,
    isClassicEditor
} from '../helpers';
import {
    editorMode,
    mobileMode,
    previewMode,
    siteMode
} from './constants';
var WidgetEnvironment = /** @class */ (function() {
    function WidgetEnvironment(options) {
        var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m, _o, _p, _q;
        this.entry = 'Widget';
        var host = options.host,
            controllerOptions = options.controllerOptions,
            predefined = options.predefined,
            query = options.query;
        this.isSSR = (_a = predefined === null || predefined === void 0 ? void 0 : predefined.isSSR) !== null && _a !== void 0 ? _a : controllerOptions === null || controllerOptions === void 0 ? void 0 : controllerOptions.isSSR;
        this.isRTL = (_b = predefined === null || predefined === void 0 ? void 0 : predefined.isRTL) !== null && _b !== void 0 ? _b : controllerOptions === null || controllerOptions === void 0 ? void 0 : controllerOptions.isRTL;
        this.language = (_c = predefined === null || predefined === void 0 ? void 0 : predefined.language) !== null && _c !== void 0 ? _c : controllerOptions === null || controllerOptions === void 0 ? void 0 : controllerOptions.language;
        this.multilingual =
            (_d = predefined === null || predefined === void 0 ? void 0 : predefined.multilingual) !== null && _d !== void 0 ? _d : controllerOptions === null || controllerOptions === void 0 ? void 0 : controllerOptions.multilingual;
        this.appDefinitionId =
            (_e = predefined === null || predefined === void 0 ? void 0 : predefined.appDefinitionId) !== null && _e !== void 0 ? _e : controllerOptions === null || controllerOptions === void 0 ? void 0 : controllerOptions.appDefinitionId;
        this.widgetId = (_f = predefined === null || predefined === void 0 ? void 0 : predefined.widgetId) !== null && _f !== void 0 ? _f : controllerOptions === null || controllerOptions === void 0 ? void 0 : controllerOptions.widgetId;
        this.isMobile = (_g = predefined === null || predefined === void 0 ? void 0 : predefined.isMobile) !== null && _g !== void 0 ? _g : (host === null || host === void 0 ? void 0 : host.formFactor) === mobileMode;
        this.isEditor = (_h = predefined === null || predefined === void 0 ? void 0 : predefined.isEditor) !== null && _h !== void 0 ? _h : (host === null || host === void 0 ? void 0 : host.viewMode) === editorMode;
        this.isPreview = (_j = predefined === null || predefined === void 0 ? void 0 : predefined.isPreview) !== null && _j !== void 0 ? _j : (host === null || host === void 0 ? void 0 : host.viewMode) === previewMode;
        this.isViewer = (_k = predefined === null || predefined === void 0 ? void 0 : predefined.isViewer) !== null && _k !== void 0 ? _k : (host === null || host === void 0 ? void 0 : host.viewMode) === siteMode;
        this.isEditorX =
            (_l = predefined === null || predefined === void 0 ? void 0 : predefined.isEditorX) !== null && _l !== void 0 ? _l : isEditorX(host === null || host === void 0 ? void 0 : host.style.styleParams);
        this.isADI = (_m = predefined === null || predefined === void 0 ? void 0 : predefined.isADI) !== null && _m !== void 0 ? _m : isADI(query);
        this.isClassicEditor =
            (_o = predefined === null || predefined === void 0 ? void 0 : predefined.isClassicEditor) !== null && _o !== void 0 ? _o : isClassicEditor(query);
        this.isCssPerBreakpoint =
            (_p = predefined === null || predefined === void 0 ? void 0 : predefined.isCssPerBreakpoint) !== null && _p !== void 0 ? _p : !!(host === null || host === void 0 ? void 0 : host.usesCssPerBreakpoint);
        this.dimensions = (_q = predefined === null || predefined === void 0 ? void 0 : predefined.dimensions) !== null && _q !== void 0 ? _q : {
            height: host === null || host === void 0 ? void 0 : host.dimensions.height,
            width: host === null || host === void 0 ? void 0 : host.dimensions.width,
        };
    }
    return WidgetEnvironment;
}());
export {
    WidgetEnvironment
};
